src/clinical/clinical-service.ts

```
@@ -128,3 +134,68 @@ export const updateDonorStats = async (donorId: number, coreCompletionOverride:
 
   return await donorDao.update(updatedDonor);
 };
+
+export async function getClinicalData(programId: string) {
+  if (!programId) throw new Error('Missing programId!');
+
+  const start = new Date().getTime() / 1000;
+
+  const donors = await getDonors(programId);
+
+  // collect all records
+  const recordsMap: any = {};
+  donors.forEach(d => {
+    Object.values(ClinicalEntitySchemaNames).forEach(entity => {
+      let clincialInfoRecords;
+      if (entity === ClinicalEntitySchemaNames.REGISTRATION) {
+        clincialInfoRecords = generateSampleRegistrationDataForDonor(d);
+      } else {
+        clincialInfoRecords = getClinicalEntitiesFromDonorBySchemaName(d, entity);
+      }
+      recordsMap[entity] = _.concat(recordsMap[entity] || [], clincialInfoRecords);
+    });
+  });
+
+  // map into object ready for api processing
+  const schemasWithFields = dictionaryManager.instance().getSchemasWithFields();
+  const data = Object.entries(recordsMap)
+    .map(([entityName, records]) => {
+      if (isEmpty(records)) return undefined;
+
+      const relevantSchemaWithFields = schemasWithFields.find(s => s.name === entityName);
```

this should throw an error if the result is undefined

---

so we can detect the issue